src/main/jbake/content/blog/2016/07/05/scaling-hawkular-alerting.adoc

```diff
@@ -0,0 +1,108 @@
+= Scaling stateful services: an example in Hawkular Alerting
+Lucas Ponce
+2016-07-05
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, alerts, scaling, partition, distributed, clustering
+
+Today stateless services architectures have the benefit to simplify the design on distributed scenarios.
+With no state in the logic and a common backend, scaling could be implemented with simple additions of new nodes in the topology and distributing incoming requests with load balancer solutions.
+
+But a stateless design can not be applied to every scenario, there are computational problems that have a stateful behaviour by nature.
+
+Scaling distributed systems with stateful scenario is both a complex but thrilling task for a software engineer.
+
+In this post we are going to expose some highlights of the design included in Hawkular Alerting.
+
+== Hawkular Alerting
+
+Hawkular Alerting is the component responsible to detect behaviours on the hawkular system by defining rules.
+These rules can generate events or alerts that can also be used in additional rules to detect complex scenarios and to respond with customized actions.
+
+This highlevel process is stateful by nature in the sense that rules might depend on the past and future data received.
+
+== Scaling: fault tolerance vs distributed computation
+
+A first (a non trivial) question is what means 'scaling' in our problem domain.
```

Comment:
Remove 'a' from within the brackets.
Suggestion: "is what is meant by 'scaling'"